plugins/mongo_db_plugin/mongo_db_plugin.cpp

```diff
@@ -379,6 +544,59 @@ void mongo_db_plugin_impl::_process_irreversible_block(const signed_block& block
    ++processed;
 }
 
+void mongo_db_plugin_impl::_process_irreversible_block(const signed_block& block)
+{
+   using namespace bsoncxx::types;
+   using namespace bsoncxx::builder;
+   using bsoncxx::builder::basic::kvp;
+   using bsoncxx::builder::stream::document;
+   using bsoncxx::builder::stream::open_document;
+   using bsoncxx::builder::stream::close_document;
+   using bsoncxx::builder::stream::finalize;
+
+
+   auto blocks = mongo_conn[db_name][blocks_col]; // Blocks
+   auto trans = mongo_conn[db_name][trans_col]; // Transactions
+
+   const auto block_id = block.id();
+   const auto block_id_str = block_id.str();
+
+   auto now = std::chrono::duration_cast<std::chrono::milliseconds>(
+         std::chrono::microseconds{fc::time_point::now().time_since_epoch().count()});
+
+   auto ir_block = find_block(blocks, block_id_str);
+
+   document update_block{};
+   update_block << "$set" << open_document << "pending" << b_bool{false}
+               << "updatedAt" << b_date{now}
+               << close_document;
+
+   blocks.update_one(document{} << "_id" << ir_block.view()["_id"].get_oid() << finalize, update_block.view());
+
+   for (const auto& packed_trx : block.input_transactions) {
```

Comment:
this seems to only mark the user generated transactions as irreversible.

When will deferred transactions be marked irreversible?  Should this go through the region/cycle/shard summaries on the irreversible block to derive transaction id's to mark irreversible?